internal/daemon/daemon.go

```diff
@@ -579,19 +579,25 @@ func (d *Daemon) State() *state.State {
 	}
 
 	state := &state.State{
-		Context:        d.shutdownCtx,
-		ReadyCh:        d.ReadyChan,
-		ShutdownDoneCh: d.shutdownDoneCh,
-		OS:             d.os,
-		Address:        d.Address,
-		Name:           d.Name,
-		Endpoints:      d.endpoints,
-		ServerCert:     d.ServerCert,
-		ClusterCert:    d.ClusterCert,
-		Database:       d.db,
-		Remotes:        d.trustStore.Remotes,
-		StartAPI:       d.StartAPI,
-		Stop:           d.stop,
+		Context:     d.shutdownCtx,
+		ReadyCh:     d.ReadyChan,
+		OS:          d.os,
+		Address:     d.Address,
+		Name:        d.Name,
+		Endpoints:   d.endpoints,
+		ServerCert:  d.ServerCert,
+		ClusterCert: d.ClusterCert,
+		Database:    d.db,
+		Remotes:     d.trustStore.Remotes,
+		StartAPI:    d.StartAPI,
+		Stop: func() (exit func(), stopErr error) {
+			err := d.stop()
```

Comment:
I was thinking more that d.stop() would return the exit function.

Reply:
Hm, I'm not sure about that. Technically with this approach, and as you've mentioned already in your comment [here](https://github.com/canonical/microcluster/pull/95#discussion_r1530024835), the daemon doesn't even need to directly send to the `d.shutdownDoneCh` as it can just immediately return after calling `d.stop()`. Only the `state` needs to do that in the API, so having an exit function serves no purpose for the `Daemon` itself. Plus, returning a `func()` from a `sync.OnceValues` sounds like a recipe for confusion.

Reply:
We also don't need to make `d.shutdownDoneCh` buffered anymore either, since the `state` will only ever call `exit()` once, and then every other time will return early because the context is already cancelled.

Reply:
To add onto this isn't the `stopErr` var now completely ignored?